### PR TITLE
Add small tile size for fp16/fp32 and NN layout

### PR DESCRIPTION
**Pull request by zjing14**

### CODE REVIEW

File: device_operation/include/device_gemm_xdl.hpp [hunk 1]
```diff
@@ -12,6 +12,8 @@
 #include "tensor_descriptor_helper.hpp"
 #include "gridwise_gemm_xdlops_v2r3.hpp"
 
+#define USE_M_N_Padding 1
```
Comment (asroy) [inline device_operation/include/device_gemm_xdl.hpp:15]: Please use a template argument to control this.


See ConvForwardSpecialization for reference
https://github.com/ROCmSoftwarePlatform/composable_kernel/blob/823657ed120144943b7db87c07fe3e647128db56/device_operation/include/device_conv2d_fwd_xdl_c_shuffle_nhwc_kyxc_nhwk.hpp#L122-L123

**Review by asroy — CHANGES_REQUESTED** [review]

**Comment by asroy** [comment]
Does this PR improve performance of small GEMM?

**Comment by zjing14** [comment]
> Does this PR improve performance of small GEMM?

Probably for extremely small Gemm. But this PR is mainly for supporting small M = 16, 32.

**Comment by asroy** [comment]
I see some perf drop with this PR in 4K GEMM..

**Comment by zjing14** [comment]
> I see some perf drop with this PR in 4K GEMM..

Which configs? I will check if it is due to M/N padding.

**Review by asroy — APPROVED** [review]

**Pull request merged**